src/types/profile-derived.js

```diff
@@ -107,3 +107,15 @@ export type MarkerTiming = {
 export type MarkerTimingRows = Array<MarkerTiming>;
 
 export type StackType = 'js' | 'native' | 'unsymbolicated';
+
+export type GlobalTrack =
+  | {| +type: 'process', +pid: Pid, +mainThreadIndex: ThreadIndex | null |}
+  | {| +type: 'screenshots' |};
+
+export type LocalTrack =
+  | {| +type: 'thread', +threadIndex: ThreadIndex |}
+  | {| +type: 'network', +threadIndex: ThreadIndex |}
+  | {| +type: 'memory', +threadIndex: ThreadIndex |};
```

Comment:
I'm not sure this is useful that these types are read only. You could make them not readonly here, but force them read only in the state by using the `$ReadOnly` utility type.

But it's fine to leave it like this now and change it later if we need it.

Reply:
I'd prefer to default to read only when we can, and only loosen up when needed later.